ui/page/root/home_page.go

```diff
@@ -276,7 +286,8 @@ func (hp *HomePage) HandleUserInteractions() {
 			case values.String(values.StrWallets):
 				pg = hp.walletSelectorPage
 			case values.String(values.StrTrade):
-				pg = NewTradePage(hp.Load)
+				hp.ensureDEXIsReady()
+				pg = NewTradePage(hp.Load, hp.dexCore)
```

Comment:
This is repeated. Let's have a method that takes item.Title and returns the page to navigate to.